### ISSUE TITLE
span<> .cbegin() not equaluvlant to std::cbegin() mismatched types

### ISSUE THREAD

**Issue by brucebob** [Microsoft/GSL#252]
```c++
using UCharSpan = gsl::span<unsigned char>;
void foo(UCharSpan sp)
{

    UCharSpan::const_iterator it = sp.cbegin(); // compiles

    // fails to compile since it's a gsl::contiguous_span_iterator
    UCharSpan::const_iterator it = std::cbegin(sp); 

}
```

Also you can not compare against the two types as well.

**Comment by neilmacintosh** [comment]
@brucebob thanks for raising this issue! I've updated the implementation for `span::iterator` and `span::const_iterator` so that they should behave correctly now, so I'll close out the issue. I added some tests, but if you feel there's still some coverage missing please feel free to offer a PR or open a new issue.

**Comment by neilmacintosh** [comment]
Sorry...and for the record the relevant PR with the fix was #320.
